src/components/input/PasswordInput.vue

```diff
@@ -1,28 +1,44 @@
 <script>
 import { defineComponent } from 'vue';
-import VisibilityIcon from '../../assets/icons/eye.svg';
+
+import BaseInput from '@/components/base/BaseInput.vue';
+import VisibilityIcon from '@/assets/icons/eye.svg';
 
 export default defineComponent({
   components: {
+    BaseInput,
     VisibilityIcon,
   },
-  props: {
-    isPasswordVisible: {
-      type: Boolean,
-      required: true,
+  data() {
+    return {
+      isPasswordVisible: false,
+    };
+  },
+  methods: {
+    togglePasswordVisibility() {
+      this.isPasswordVisible = !this.isPasswordVisible;
+    },
+  },
+  computed: {
+    computedType() {
+      if (this.isPasswordVisible) return 'text';
+      else return 'password';
     },
   },
 });
 </script>
 
 <template>
-  <button
-    type="button"
-    class="password-visibility-button"
-    :class="{ 'password-visibility-button--shown': isPasswordVisible }"
-  >
-    <VisibilityIcon />
-  </button>
+  <BaseInput :type="computedType">
```

Comment:
Like with the `DateInput` - as a good practice I'd suggest to implement a v-model here as well (in a nested fashion - so that passing the value prop here would bind it to the `value` of `BaseInput` and also handling the `BaseInput`'s event manually by re-emitting + applying any logic (if there is one).
This way you could explicitly say that these components emit such events and prop information would give us also more help on how to use the component.